sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/model/Bid.java

```diff
@@ -156,6 +157,28 @@ public Bid withoutAnnotation(String annotation) {
     }
   }
 
+  @Override
+  public boolean equals(Object otherObject) {
+    if (this == otherObject) {
+      return true;
+    }
+    if (otherObject == null || getClass() != otherObject.getClass()) {
+      return false;
+    }
+
+    Bid other = (Bid) otherObject;
+    return auction == other.auction
+        && bidder == other.bidder
+        && price == other.price
+        && dateTime == other.dateTime
```

Comment:
this made me double check the type of `dateTime` field. I think it's less error-prone to use `Objects.equals()` even for primitive types